From aadaae9305a96da301e1466dbe93b85d7e89b2f9 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hongjiu.lu@intel.com>
Date: Tue, 5 Feb 2013 21:47:40 +0330
Subject: [PATCH 09/18] Support detection on btrfs software raid (#906847)

Update btrfs support to include detecting systems where the btrfs
filesystem sits on top of software raid.

Resolves: rhbz#906847
---
 os-prober | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os-prober b/os-prober
index ca8f5ab..f5052af 100755
--- a/os-prober
+++ b/os-prober
@@ -149,7 +149,7 @@ for partition in $(partitions); do
 
 	# do btrfs processing here; both mounted and unmounted will
 	# be handled by 50mounted-tests so we can do a subvol only once.
-	type=$(blkid -o value -s TYPE $mapped)
+	type=$(blkid -o value -s TYPE $mapped || true)
 	if [ "$type" = btrfs ]; then
 		uuid=$(blkid -o value -s UUID $mapped)
 		if grep -q "^$uuid" "$OS_PROBER_TMP/btrfs-vols" ; then
-- 
2.5.5

